Makefile

```
@@ -75,8 +75,8 @@ autoformat: format-ruff format-codespell  ## Run all automatic formatters
 
 .PHONY: format-ruff
 format-ruff:  ##- Automatically format with ruff
-	ruff format $(SOURCES)
 	ruff check --fix $(SOURCES)
+	ruff format $(SOURCES)
```

Why?

---

Because sometimes the code changes that `check` makes are actually not `format`ted.

---

I'm uncertain how I feel about this given that `ruff check --fix` will exit nonzero if there are leftover unfixable linting issues. If we have enough time in overflow today let's ask the team how they feel about adding `--exit-zero` to the `ruff check` command so we still format even if there are remaining lint violations (the original reason I formatted before checking).

The only potential downside I can think of is that `make format-ruff` will exit with success even if there are remaining lint errors, but I'm okay with that personally because every use case I can think of where we'd want to check the result we'd probably be using `make lint-ruff` instead.

---

Ok interesting.  If we do it this way, we'll always format and exit with a correct return code:

```suggestion
        success=true
        ruff check --fix $(SOURCES) || success=false
	ruff format $(SOURCES)
	$success || exit 1
```

---

Sure, sounds good